kbpagesd/main.go

```diff
@@ -38,8 +39,24 @@ func init() {
 }
 
 func newLogger(isCLI bool) (*zap.Logger, error) {
-	var loggerConfig zap.Config
+	// In keybase/client/go/logger fd 2 is closed. To make sure our logger can
+	// log to stderr, duplicate the fd beforehand. Apparently it's important to
+	// call this function before any keybase/client/go/logger logging is set
+	// up.
+	stderr, err := stderrutils.DupStderr()
+	if err != nil {
+		panic(err)
+	}
+
+	// Zap loggers use os.Stderr by default. We could pass in stderr by making
+	// more boilerplate, but there's not much else we need from those. So
+	// override os.Stderr temporarily as a hack to inject stderr to the zap
+	// logger.
+	originalStderr := os.Stderr
+	os.Stderr = stderr
```

Comment:
It can, but as described in comment, would require more boilerplates, since we'd use lower-layer APIs instead of the ones used here.

This is only called when initializing, so I don't think goroutines matter?

In the long term, we'd likely switch to the approach with more boilerplates since we need to redirect logs to logstash. I'll swap this out when that happens if that's OK with you.

Reply:
All right, but then leave a TODO please.

> This is only called when initializing, so I don't think goroutines matter?

What if other goroutines use `os.Stderr` concurrently with this reassignment?

Reply:
> All right, but then leave a TODO please.

Sure will do.

> What if other goroutines use os.Stderr concurrently with this reassignment?

This is called at the beginning of `main()`, where no other goroutine has been spawned yet.